lib/dangermattic/plugins/milestone_checker.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin for performing checks on a milestone associated with a pull request.
+  class MilestoneChecker < Plugin
+    DEFAULT_WARNING_DAYS = 5
+
+    # Checks if the pull request is assigned to a milestone.
+    def check_milestone_set(fail_on_error: false)
+      return unless milestone.nil?
+
+      message = 'PR is not assigned to a milestone.'
+      if fail_on_error
+        failure(message, sticky: false)
+      else
+        warn(message, sticky: false)
+      end
+    end
+
+    # Checks if the pull request's milestone is due to finish within a certain number of days.
+    #
+    # @param warning_days [Integer] Number of days to warn before the milestone due date (default: DEFAULT_WARNING_DAYS).
+    # @param if_no_milestone [Symbol] Action to take if the pull request is not assigned to a milestone. Possible values:
+    #                 - :warn (default): Reports a warning.
+    #                 - :error: Reports an error.
+    #                 - :none or nil: Takes no action.
+    def check_milestone_due_date(warning_days: DEFAULT_WARNING_DAYS, if_no_milestone: :warn)
```

Comment:
Nitpick: Given that the behavior now allows for reporting an error, too, `warning_days` is not 100% aligned as a name.

What do you think of `days_threshold` or `days_cutoff`? I don't love either, but I'm trying to describe the "threshold of days left below which to ~warn~ report".

Reply:
Indeed; I've tried `days_before_due: DEFAULT_DAYS_THRESHOLD` (cc67ee757c47d5eaa56eaa5c0949dc0d61be3bdb). I think it is kinda hard to have a really concise name, the parameter description should help 🤷